plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialSwitchable.java

```diff
@@ -36,218 +45,85 @@ public static boolean describes(ObjectTag material) {
-        // More specifically, this returns whether:
-        // - a Powerable material (like pressure plates) is activated
-        // - an Openable material (like doors) is open
-        // - a dispenser is powered and should dispense its contents
-        // - a daylight sensor is inverted (detects darkness instead of light)
-        // - a lightable block is lit
-        // - a piston block is extended
-        // - an end portal frame has an ender eye in it
-        // - a hopper is NOT being powered by redstone
-        // - a sculk_shrieker can summon a warden
-        // -->
-        PropertyParser.registerStaticTag(MaterialSwitchable.class, ElementTag.class, "switched", (attribute, material) -> {
-            return new ElementTag(material.getState());
-        });
-    }
-
-    public boolean isPowerable() {
-        return material.getModernData() instanceof Powerable;
-    }
-
-    public boolean isOpenable() {
-        return material.getModernData() instanceof Openable;
-    }
-
-    public boolean isDisepnser() {
-        return material.getModernData() instanceof Dispenser;
-    }
-
-    public boolean isDaylightDetector() {
-        return material.getModernData() instanceof DaylightDetector;
-    }
-
-    public boolean isLightable() {
-        return material.getModernData() instanceof Lightable;
-    }
-
-    public boolean isPiston() {
-        return material.getModernData() instanceof Piston;
-    }
-
-    public boolean isEndFrame() {
-        return material.getModernData() instanceof EndPortalFrame;
-    }
-
-    public boolean isHopper() {
-        return material.getModernData() instanceof Hopper;
-    }
-
-    public boolean isSculkShrieker() {
-        return NMSHandler.getVersion().isAtLeast(NMSVersion.v1_19) && material.getModernData() instanceof SculkShrieker;
-    }
-
-    public Powerable getPowerable() {
-        return (Powerable) material.getModernData();
-    }
-
-    public Openable getOpenable() {
-        return (Openable) material.getModernData();
-    }
-
-    public Dispenser getDispenser() {
-        return (Dispenser) material.getModernData();
-    }
-
-    public DaylightDetector getDaylightDetector() {
-        return (DaylightDetector) material.getModernData();
-    }
-
-    public Piston getPiston() {
-        return (Piston) material.getModernData();
-    }
-
-    public Lightable getLightable() {
-        return (Lightable) material.getModernData();
+    public MaterialSwitchable(MaterialTag material) {
+        super(material);
     }
 
-    public EndPortalFrame getEndFrame() {
-        return (EndPortalFrame) material.getModernData();
-    }
-
-    public Hopper getHopper() {
-        return (Hopper) material.getModernData();
-    }
-
-    /*public SculkShrieker getSculkShrieker() { // TODO: 1.19
-        return (SculkShrieker) material.getModernData();
-    }*/
-
-    public boolean getState() {

```

Comment:
considering getState was used in other code, it should probably be left as-is, same as setState was